x-pack/legacy/plugins/lens/public/xy_visualization_plugin/xy_expression.tsx

```diff
@@ -100,6 +100,33 @@ function getIconForSeriesType(seriesType: SeriesType): IconType {
   return chartTypeIcons.find(chartTypeIcon => chartTypeIcon.id === seriesType)!.iconType;
 }
 
+function getTickFormatter(layers: LayerArgs[], data: LensMultiTable) {
+  const isDateXAxis = Object.values(layers).every(layer => layer.xScaleType === 'time');
+
+  function ofAllXValues(fn: (...args: number[]) => number) {
+    return fn.apply(
+      undefined,
+      Object.values(layers).map(
+        layer =>
+          data.tables &&
+          data.tables[layer.layerId] &&
+          fn.apply(
+            undefined,
+            data.tables[layer.layerId].rows.map(row => row[layer.xAccessor] as number)
+          )
+      )
+    );
+  }
+
+  if (!isDateXAxis) {
+    return (x: string) => x;
+  } else {
+    const minDate = ofAllXValues(Math.min);
+    const maxDate = ofAllXValues(Math.max);
+    return niceTimeFormatter([minDate, maxDate]);
```

Comment:
This logic doesn't really make sense to me- the extent of the data should be unrelated to the formatting. I would expect that based on the interval in the data source, the formatter would change precisions- regardless of how much data is returned

Reply:
You are right, this is more like a rule of thumb which takes advantage of the fact that the density of data points shown on the screen doesn't change that much - because of that, the interval and the extent of the data correlates.

The rule behind `niceTimeFormatter` is
```
  if (days > 30) {
    return 'yyyy-MM-dd';
  }
  if (days > 7 && days <= 30) {
    return 'MMMM dd';
  }
  if (days > 1 && days <= 7) {
    return 'MM-dd HH:mm';
  }
  return 'HH:mm:ss';
```

Which I think is good enough as a temporary solution for beta. In the long run this should be configurable in the chart itself.